common-components/schemas/type/money.md

```diff
@@ -0,0 +1,36 @@
+# Money
+
+The `Money` type represents an amount of money in a specified currency.
+
+## Schema
+
+A `Money` has two fields:
```

Comment:
I feel like we should have some way to define this in a programmatically consumable format, and have that be the source of truth. This content is largely redundant with the content in the proto.

That format can translate itself into the proto, as well as the .md file. 

I primarily worry about the drift in different representations if we don't have some tooling to help here.

Reply:
I think that's a fair point.  A couple counter-points:

* We don't expect the number of common components to be large enough that manual maintenance is intractable, right?  Especially since we don't expect common components to change over time?
* I actually quite deliberately made the canonical definitions IDL-agnostic, treating formal IDL definitions as implementation details.

But those aside, sure - I don't actually feel too strongly about this.  How would you recommend we canonically define common components, if not prose + bullet points?  If you know of some appropriate formal syntax/DSL for defining such things, great.

Or we could just say that protos are canonical, and other IDLs are secondary?

Reply:
I was thinking we could have our own yaml format, or use JSONSchema and translate it.

But I'm fine with merging these first couple as a reference point, and then we can write a script to do the translation and generation. I think it's *pretty* straightforward, so I'd be happy to pick that up once we have an example.

> We don't expect the number of common components to be large enough that manual maintenance is intractable, right? Especially since we don't expect common components to change over time?

Sure, but I think any manual cost we can reduce would be great. it is low-hanging fruit since the content is roughly the same. 

> I actually quite deliberately made the canonical definitions IDL-agnostic, treating formal IDL definitions as implementation details.

I think this further emphasizes that code-genning these would be better.

Reply:
Unfortunate JSON isn't sufficient because it doesn't have a way to specify e.g. `int32` and `int64`; it just has `number`.  My first thought was to use TypeScript interfaces like some of the AIPs do, but the lack of distinct numeric types system made it a non-starter.

And sure, we could come up with our own DSL for this.  It's a principled approach, but also seems a bit yak-shavey.  And how different will it really end up looking than protobuf?